core/lib/config/src/configs/vm_runner.rs

```diff
@@ -27,6 +30,10 @@ pub struct BasicWitnessInputProducerConfig {
     pub window_size: u32,
     /// All batches before this one (inclusive) are always considered to be processed.
     pub first_processed_batch: L1BatchNumber,
+
+    #[serde(skip)]
+    // manually added when parsing from env because `envy` doesn't support `serde(flatten)`
+    pub experimental_vm: ExperimentalVmConfig,
```

Comment:
I guess instead of embedding `ExperimentalVmConfig` into component configs, we may require 2 configs as input for components. Since we're running infra in the distributed mode, we can still override the experimental VM config on a per-component basis.

Reply:
What would be the benefits? IMO, having VM config embedded into component configs leads to better config structure, especially for file-based configs.

Reply:
Less special handling (here we have to manually add it), less duplication in configuration, easier to remove later (I consider this to be temporary config).

Reply:
Yes, I also consider this config temporary, that's why it's marked as experimental. Suppose we integrate the new VM into the API server and EN state keeper as well. I think it's reasonable to expect that we would have the VM switch configurable in 3 places for some time. Do you think it would be better to have a single config with "mode in BWIP" / "mode in state keeper" / "mode in API server sandbox" fields?

Reply:
No, I think it's better to have components accept a tuple: `(component_config, experimental_vm_mode_config)`.
This way, we can gradually add this config to the components, and as long as the configuration is provided for some _env_ (stage/testnet/mainnet), enabling it will be a no-op. If a divergence is found and we need to rollback -- changing the config in one place will change it back everywhere (which IMHO makes sense, because if there is a divergence, likely it will be present in all the components that will process the same batch).

If we want to force a different value for a particular component, we can add override on the component level.

Reply:
With the VM playground component, there are no more embedded configs (for now).